src/typer_core.erl

```diff
@@ -51,10 +51,10 @@
          %% Files in 'fms' are compilable with option 'to_pp'; we keep them
          %% as {FileName, ModuleName} in case the ModuleName is different
          fms = [] :: [{file:filename(), module()}],
-         ex_func = map_dict_new() :: map_dict(),
-         record = map_dict_new() :: map_dict(),
-         func = map_dict_new() :: map_dict(),
-         inc_func = map_dict_new() :: map_dict(),
+         ex_func = maps:new() :: map(),
+         record = maps:new() :: map(),
+         func = maps:new() :: map(),
+         inc_func = maps:new() :: map(),
```

Comment:
Can we be more precise here, using stuff like `#{atom() => …}` instead of just the generic `map()`?